tests/integration/test_extensions.py

```diff
@@ -354,3 +355,34 @@ def test_hard_exit(tmpdir, py_finalize):
 
     # THEN
     # No assertions, just check that the program exits without error
+
+
+@pytest.mark.skipif(
+    sys.platform == "darwin", reason="Test requires a linker that supports -rpath"
+)
+def test_dlopen_with_rpath(tmpdir, monkeypatch):
+    # GIVEN
+    output = Path(tmpdir) / "test.bin"
+    extension_name = "sharedlibs"
+    extension_path = tmpdir / extension_name
+    shutil.copytree(TEST_RPATH_EXTENSION, extension_path)
+    subprocess.run(
+        [sys.executable, str(extension_path / "setup.py"), "build_ext", "--inplace"],
+        check=True,
+        cwd=extension_path,
+        capture_output=True,
+    )
+
+    # WHEN
+    with monkeypatch.context() as ctx:
+        ctx.setattr(sys, "path", [*sys.path, str(extension_path)])
+        from ext import hello_world  # type: ignore
+
+        try:
+            hello_world()
+        except RuntimeError:
+            pytest.skip("Test requires a linker that supports -rpath")
```

Comment:
Hm - is this still necessary given the `@pytest.mark.skipif` above for Darwin? What platform do we support that's not Darwin but where `hello_world()` would fail?

I'm a bit worried that this would mask real errors (turning them into skips instead of test failures) if we screw something up in the future.

Reply:
This is needed for old versions of the linker like the ones in the many Linux images.

Reply:
Ah right, I remember now - I just didn't connect the dots.

OK, let's leave it then. I've been thinking about restructuring this CI to work like PyStack's, using cibuildwheel to build the wheels but _not_ test them, and downloading the built wheels onto more modern containers and testing them there (both to reintroduce a bit more parallelism, and to prove that the wheels are working correctly on newer distros).

If I make that change, I should be able to remove this at that point.

Reply:
> I've been thinking about restructuring this CI to work like PyStack's, using cibuildwheel to build the wheels but _not_ test them, and downloading the built wheels onto more modern containers and testing them there (both to reintroduce a bit more parallelism, and to prove that the wheels are working correctly on newer distros).

Actually... I've investigated this a bit and decided not to do it, because it _won't_ introduce more parallelism - at least, not without switching back to multiple matrix jobs. As things stand today, I'd need to make a test_on_ubuntu job that depends on our build_linux_wheels job, but the build_linux_wheels job is a matrix job that doesn't finish until all its sub-jobs are done, so we wouldn't wind up testing any Linux wheels until we're done building all of them across all architectures, which is actually probably worse than the current behavior.